lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -764,6 +769,22 @@ listPools spl = liftHandler (map mkApiStakePool <$> listStakePools spl)
             (ApiT pool)
             (StakePoolMetrics (Quantity $ fromIntegral stake) blocks)
 
+joinPool
+    :: StakePoolLayer IO
```

Comment:
Should the `StakePoolLayer` (whose perhaps questionable existence I'm to blame for) really be involved?

Won't `join-` and `quit-` stake-pool be very tightly coupled with everything wallet-related? (get private key, have a transaction layer)

Only thing I see the `StakePoolLayer` being good for in #896, is verifying that a stake-pool exists, which we might as well use the `NetworkLayer` for, instead of querying the `Pool.DB`.

Reply:
yes, I already changed to `PutStakePool` and `DeleteStakePool` in `Cardano.Wallet.Api` as we mimick there REST API verbs